cli/src/cmd/forge/verify/provider.rs

```diff
@@ -15,10 +15,10 @@ pub trait VerificationProvider {
     /// [`VerifyArgs`] are valid to begin with. This should prevent situations where there's a
     /// contract deployment that's executed before the verify request and the subsequent verify task
     /// fails due to misconfiguration.
-    async fn preflight_check(&self, args: VerifyArgs) -> eyre::Result<()>;
+    async fn preflight_check(&mut self, args: VerifyArgs) -> eyre::Result<()>;
 
     /// Sends the actual verify request for the targeted contract.
-    async fn verify(&self, args: VerifyArgs) -> eyre::Result<()>;
+    async fn verify(&mut self, args: VerifyArgs) -> eyre::Result<()>;
```

Comment:
I get why having a cached `CachedEntry` for the EtherscanProvider is useful; therefore, we need the `&mut` here.

I don't want to obfuscate this via `Arc<Mutext>`, but we need to document why the mut is here.

but imo for verification, we perhaps don't really need this as the perf can be neglected here entirely.

so I wouldn't actual bother to cache anything.

Reply:
tbh i'd prefer arc mutex to multiple reads

Reply:
right,
after thinking about this again, storing the elements is actually safer since we don't need to run anything that could potential fail after deploying the contract.

so lgtm